leshan-client-core/src/main/java/org/eclipse/leshan/client/servers/ServersInfoExtractor.java

```diff
@@ -112,6 +122,16 @@ public static ServersInfo getInfo(Map<Integer, LwM2mObjectEnabler> objectEnabler
                         info.clientCertificate = getClientCertificate(security);
                         info.serverCertificate = getServerCertificate(security);
                         info.privateKey = getPrivateKey(security);
+                    } else if (info.secureMode == SecurityMode.OSCORE) {
```

Comment:
as there is no OSCORE securityMode maybe we can just look if the resource OSCORE Security Mode(17) is set.

If this is set you get the right index for the FIXME below :)

Reply:
Yes, I am no longer checking the security mode here to decide what to do but rather if the link from the Security object has been set. I am also using the link from the Security object to find the OSCORE object here. Done in commit 3e7db26f5131ad4ef4622ae864e69d5796b3eefa